server/balancer.go

```diff
@@ -417,3 +418,75 @@ func (db *defaultBalancer) Balance(cluster *clusterInfo) (*score, *balanceOperat
 
 	return nil, bop, errors.Trace(err)
 }
+
+// replicaBalancer is used to balance active replica count.
+type replicaBalancer struct {
+	cfg             *BalanceConfig
+	region          *metapb.Region
+	leader          *metapb.Peer
+	downPeers       []*pdpb.PeerStats
+	defaultBalancer *defaultBalancer
+}
+
+func newReplicaBalancer(region *metapb.Region, leader *metapb.Peer, downPeers []*pdpb.PeerStats, cfg *BalanceConfig) *replicaBalancer {
+	return &replicaBalancer{
+		cfg:             cfg,
+		region:          region,
+		leader:          leader,
+		downPeers:       downPeers,
+		defaultBalancer: newDefaultBalancer(region, leader, cfg),
+	}
+}
+
+func (rb *replicaBalancer) addPeer(cluster *clusterInfo) (*balanceOperator, error) {
+	return rb.defaultBalancer.addPeer(cluster)
+}
+
+func (rb *replicaBalancer) removePeer(cluster *clusterInfo, downPeers []*metapb.Peer) (*balanceOperator, error) {
+	if len(downPeers) >= 1 {
+		removePeerOperator := newRemovePeerOperator(rb.region.GetId(), downPeers[0])
+		return newBalanceOperator(rb.region, newOnceOperator(removePeerOperator)), nil
+	}
+	return rb.defaultBalancer.removePeer(cluster)
+}
+
+func (rb *replicaBalancer) collectDownPeers(cluster *clusterInfo) []*metapb.Peer {
+	var downPeers []*metapb.Peer
+	for _, stats := range rb.downPeers {
+		peer := stats.GetPeer()
+		if peer == nil {
+			continue
+		}
+		store := cluster.getStore(peer.GetStoreId())
+		if store == nil {
+			continue
+		}
+		if stats.GetDownSeconds() >= rb.cfg.MaxPeerDownDuration {
```

Comment:
Each time I see Duration, I think this is time.Duration, but here is seconds.
maybe we should think a better name? @qiuyesuifeng